### PR TITLE
ssh_identify_pubkeys fixes

### PR DESCRIPTION
**Pull request by jhart-r7**
In using this module to track down where else a users SSH authorized keys could get them, I've been feeding `authorized_keys` files through this module and noticed and fixed a few problems:
- `KEY_DIR` and `KEY_PATH` don't expand if they have symbolic values in them (`~/foo/bar.txt`).  This fixes that, I think.
- Public keys that also included the commands the user can run would not work.  
- Several unused (AFAIK) options

### CODE REVIEW

**Comment by wvu-r7** [comment]
Used the module just yesterday. I'll grab it.

**Comment by stufus** [comment]
argh sorry I missed these in https://github.com/rapid7/metasploit-framework/pull/6318 
:-(

**Comment by jhart-r7** [comment]
@wvu-r7, please ping me if I can help get this landed.  Thanks!

**Comment by wvu-r7** [comment]
@jhart-r7: The changes look good, but I'm seeing if there's a better way to handle tilde expansion proactively. Really glad you implemented it, though. It's been a wanted feature, but no one has been particularly vocal about it.

**Comment by wvu-r7** [comment]
And testing was smooth, FWIW. Sorry for not noting that.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb [hunk 1]
```diff
@@ -68,7 +70,11 @@ def initialize
   end
 
   def key_dir
-    datastore['KEY_DIR']
+    ::File.expand_path(datastore['KEY_DIR'])
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: Is any of this necessary when you normalize with ```File.expand_path```?

**Comment by jhart-r7** [comment]
Bah.  Merge conflicts now.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb (hunk 1 above)
Comment (jhart-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: No.

**Comment by jhart-r7** [comment]
@wvu-r7 sorry for the delay on this.  I've addressed the merge conflict and your last comment.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb [hunk 2]
```diff
@@ -84,19 +90,19 @@ def read_keyfile(file)
       keyfile = datastore['SSH_KEYFILE_B64'].unpack("m*").first
     elsif file.kind_of? Array
       keyfile = ''
-      file.each do |dir_entry|
+      file.map { |f| ::File.expand_path(f) }.each do |dir_entry|
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: I'm not sure you need ```File.expand_path``` here either. I tested without it just fine.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb [hunk 3]
```diff
@@ -84,19 +90,19 @@ def read_keyfile(file)
...
         next unless ::File.readable? dir_entry
         keyfile << ::File.open(dir_entry, "rb") {|f| f.read(f.stat.size)}
       end
     else
-      keyfile = ::File.open(file, "rb") {|f| f.read(f.stat.size)}
+      keyfile = ::File.open(::File.expand_path(file), "rb") {|f| f.read(f.stat.size)}
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: Or here.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb [hunk 4]
```diff
@@ -162,8 +168,8 @@ def pull_cleartext_keys(keys)
 
   def do_login(ip, port, user)
 
-    if datastore['KEY_FILE'] and File.readable?(datastore['KEY_FILE'])
-      keys = read_keyfile(datastore['KEY_FILE'])
+    if datastore['KEY_FILE'] && File.readable?(key_file)
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: Just ```if File.readable?(keyfile)``` should be fine here.

File: lib/msf/core/opt_path.rb [hunk 5]
```diff
@@ -23,7 +27,7 @@ def valid?(value)
       if value =~ /^memory:\s*([0-9]+)/i
         return false unless check_memory_location($1)
       else
-        unless File.exist?(value)
+        unless File.exists?(File.expand_path(value))
```
Comment (wvu-r7) [inline lib/msf/core/opt_path.rb]: ```File.exists?``` is deprecated. You had it right before.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb (hunk 1 above)
Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: By "any of this," I meant the other cases of ```File.expand_path``` in the code. Didn't mean to sound snarky.

**Comment by wvu-r7** [comment]
Re https://github.com/rapid7/metasploit-framework/pull/6528#issuecomment-186045629, I was wondering if we could do this only in ```normalize```. Looks like that was the case.

File: modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb (hunk 2 above)
Comment (jhart-r7) [inline modules/auxiliary/scanner/ssh/ssh_identify_pubkeys.rb]: I believe this is now duplicated and unnecessary given my changes in lib/msf/core/opt_path.rb

**Comment by jhart-r7** [comment]
@wvu-r7, you are correct.  Since the expand_path stuff happens in `normalize` now, there is no need to redo it in the module.  Fixed.

**Comment by wvu-r7** [comment]
@jhart-r7: Ah, okay. Guess it was just leftover code. No worries.

**Comment by wvu-r7** [comment]
w00t!

**Pull request merged**